packages/odyssey-react/src/utils/focusHandling.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+type OptionalHTMLElement = HTMLElement | null;
+
+interface UseFocusHook {
+  restoreFocus: (current: OptionalHTMLElement) => void;
+  setFocus: (elem: OptionalHTMLElement) => OptionalHTMLElement;
+}
+
+const FOCUSABLE_ITEMS = [
+  "button",
+  "[href]",
+  "input",
+  "select",
+  "textarea",
+  '[tabindex]:not([tabindex="-1"])',
+];
+
+const FOCUSABLE_ITEMS_SELECTOR = FOCUSABLE_ITEMS.join(",");
+
+/**
+ * Set focus on first focusable element inside node tree
+ * @param {HTMLElement} elem - parent element that contains focusable child elements
+ * @returns {void}
+ */
+function setFocus(elem: OptionalHTMLElement): OptionalHTMLElement {
```

Comment:
rather than annotating this argument and return value as `HTMLElement | null` can we use a generic here to infer the type for us? That way we'll have better type safety.

Reply:
HI @jeffbelser-okta , can you clarify what you mean by "use a generic here to infer the type"?